### PR TITLE
Logan Wright | React Testing

### PR DESCRIPTION
**Pull request by LBWright**

### CODE REVIEW

File: src/__tests__/App.spec.js [hunk 1]
```diff
@@ -3,13 +3,74 @@ import ReactDOM from 'react-dom';
...
+  describe('handleClick()', () => {
+    it('should call "calculate" exactly one time', () => {
+      // to know how many times a function is called
+      const root = shallow(<App />);
+      const instance = root.instance();
+      const buttonName = 'logan';
+
+      instance.handleClick(buttonName);
+
+      expect(calculate).toHaveBeenCalledTimes(1);
```
Comment (jayyargh) [inline src/__tests__/App.spec.js]: I saw a lot of toBe()'s and was concerned you didn't experiment with any other test methods, but I saw this and felt better. It's fun looking through the docs and seeing different methods and thinking, "Hmmm... how can I implement that into one of my tests?" and you clearly had that a-to-b process and nailed it.

File: src/__tests__/App.spec.js [hunk 2]
```diff
@@ -3,13 +3,74 @@ import ReactDOM from 'react-dom';
...
+    });
+
+    it('should call "calculate" passing the state and buttonName', () => {
```
Comment (jayyargh) [inline src/__tests__/App.spec.js]: I can tell Luis's lecture helped give you more... direction? I guess? Either way, I like your thought process here where you're testing state changes.

File: src/__tests__/Button.spec.js [hunk 3]
```diff
@@ -5,11 +5,15 @@ import Adapter from 'enzyme-adapter-react-16';
...
-    const div = document.createElement('div');
-    ReactDOM.render(<Button />, div);
+    shallow(<Button />);
+  });
+
+  it('should render a button a value of its "name" prop', () => {
+    const root = shallow(<Button name="name" />);
+    const instance = root.instance();
+
+    expect(root.find('button').text()).toEqual('name');
```
Comment (jayyargh) [inline src/__tests__/Button.spec.js]: This is a good test to determine whether React is capable of creating a DOM element with the attribute you prescribed it or not. If this failed, you could then immediately go to the React github and create an issue.

File: src/__tests__/Display.spec.js [hunk 4]
```diff
@@ -5,11 +5,13 @@ import Adapter from 'enzyme-adapter-react-16';
...
 describe('<Display />', () => {
   it('renders without crashing', () => {
-    const div = document.createElement('div');
-    ReactDOM.render(<Display />, div);
+    shallow(<Display />);
+  });
+
+  it('should display the value of its "value" prop', () => {
+    const root = shallow(<Display value="3" />);
+    expect(root.find('.value').text()).toEqual('3');
```
Comment (jayyargh) [inline src/__tests__/Display.spec.js]: Looks good!

File: src/__tests__/Panel.spec.js [hunk 5]
```diff
@@ -5,11 +5,12 @@ import Adapter from 'enzyme-adapter-react-16';
...
-
 describe('<Panel />', () => {
   it('renders without crashing', () => {
-    const div = document.createElement('div');
-    ReactDOM.render(<Panel />, div);
+    shallow(<Panel />);
+  });
+  it('renders all Button Components', () => {
+    const wrapper = shallow(<Panel />);
+    expect(wrapper.find('Button').length).toBe(19);
```
Comment (jayyargh) [inline src/__tests__/Panel.spec.js]: Good test, ensuring that all buttons necessary for the calculator's functionality are rendering to the panel! Nice.

File: src/__tests__/Display.spec.js (hunk 4 above)
Comment (LBWright) [reply src/__tests__/Display.spec.js]: Thanks! @jayyargh

**Comment by jayyargh** [comment]
Hey Logan,

I really appreciate the thoughtfulness in your testing and your willingness to experiment with different methods and state changes. I think your code is solid and I was able to understand everything you were testing without struggle. 

I look for three things when reviewing code:

1. Initially, I check commit messages. I should be able to determine what was done/accomplished/etc. just by looking at them. A developer who commits regularly with detailed commits is invaluable on any team. Honestly, that's my biggest nitpick with this repo. Commit more: Added a test? Commit it. Something's not working? Revert to last commit. Didn't commit for several test cases and a few hours of work? That's a problem on you for not entering a simple command and utilizing your toolset and I've been there and it's certainly disappointing.

2. Functionality. All of your tests operate per expectation and do exactly what they say they do. I had no weird errors when I ran the tests, no side effects, and all green down the line. That mint green really is satisfying, isn't it?

3. Finally, consistent naming conventions, good linting, and best practices. All of which fall under readability, I guess. You excelled at this and I had no problem understanding what a variable was declared for. Everything looked straightforward and purposeful, just as code is meant to be. Nice work.

**Pull request closed without merging**